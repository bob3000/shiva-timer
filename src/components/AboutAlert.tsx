import * as React from 'react';
import { Alert, Linking, Text } from 'react-native';

interface IAboutAlertProps {
  okHandler: () => void;
  visible: boolean;
}

// tslint:disable-next-line: no-var-requires
const packageJson = require('../../package.json');
const alertText = `by Robin Kautz

Version ${packageJson.version}
`;
const website = 'http://github.com/bob3000/shiva-timer';

const AboutAlert: React.FunctionComponent<IAboutAlertProps> = (props) => {
  if (!props.visible) return null;
  return (
    <Text style={{ height: 0 }}>
      {Alert.alert('Shiva Timer', alertText, [
        { text: 'Go to Website', onPress: () => Linking.openURL(website) },
        { text: 'Close', onPress: props.okHandler },
      ])}
      ;
    </Text>
  );
};

export default AboutAlert;
