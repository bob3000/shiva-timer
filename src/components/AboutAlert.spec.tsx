import * as React from 'react';
import * as renderer from 'react-test-renderer';
import AboutAlert from './AboutAlert';

describe('AboutAlert', () => {
  it('snapshot test', () => {
    const tree = renderer
      .create(<AboutAlert okHandler={() => null} visible={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
