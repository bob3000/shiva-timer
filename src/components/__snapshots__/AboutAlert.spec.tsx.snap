// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutAlert snapshot test 1`] = `
<Text
  style={
    Object {
      "height": 0,
    }
  }
>
  ;
</Text>
`;
